src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -200,13 +200,14 @@ public override SyntaxNode ParameterDeclaration(string name, SyntaxNode type, Sy
                 initializer != null ? SyntaxFactory.EqualsValueClause((ExpressionSyntax)initializer) : null);
         }
 
-        internal static SyntaxTokenList GetParameterModifiers(RefKind refKind)
+        internal static SyntaxTokenList GetParameterModifiers(RefKind refKind, bool forFunctionPointerReturnParameter = false)
```

Comment:
i don't think this new parameter has value.  ca we just remove and respect the refkind passed in.  i.e. if somone asks for `.Out` they get `out`.  c'est la vie, that's what they asked for.

Reply:
No. When we generate the modifiers for the return type of a function pointer, `In` means `RefReadonly`. These values are the same.

Reply:
I'll add a comment that `In` == `RefReadOnly`

Reply:
I think I get it. I think I want this parameter to be more like 'bool isParameter'. In parameters we allow 'in' but not in places like a property return.

Reply:
I considered making it `isParameter`, but given that the function itself is used for generating parameter modifiers I felt it better to be specific about the scenario here. Regular return type modifiers are parsed and handled in a different way that parameter modifiers so I don't want to make the function name less specific.

Reply:
Added more to the comment. I'm keeping the parameter anme.